internal/k8s/validator.go

```diff
@@ -717,6 +727,37 @@ func (l *leader) ValidateL4RuleObj(key string, l4Rule *akov1alpha2.L4Rule) error
 	return nil
 }
 
+// ValidateL7RuleObj would do validation checks and updates the status before
+// pushing to ingestion
+func (l *leader) ValidateL7RuleObj(key string, l7Rule *akov1alpha2.L7Rule) error {
+	l7RuleSpec := l7Rule.Spec
+	refData := make(map[string]string)
+	if l7RuleSpec.BotPolicyRef != nil {
+		refData[*l7RuleSpec.BotPolicyRef] = "BotPolicy"
+	}
+
+	if l7RuleSpec.SecurityPolicyRef != nil {
+		refData[*l7RuleSpec.SecurityPolicyRef] = "SecurityPolicy"
+	}
+
+	if l7RuleSpec.TrafficCloneProfileRef != nil {
+		refData[*l7RuleSpec.TrafficCloneProfileRef] = "TrafficCloneProfile"
+	}
+	if err := checkRefsOnController(key, refData); err != nil {
+		status.UpdateL7RuleStatus(key, l7Rule, status.UpdateCRDStatusOptions{
+			Status: lib.StatusRejected,
+			Error:  err.Error(),
+		})
+		return err
+	}
+	// No need to update status of l4rule object as accepted since it was accepted before.
```

Comment:
might want to change l4rule to l7rule here

Reply:
changed